osquery/tables/system/windows/appcompat_shims.cpp

```diff
@@ -0,0 +1,92 @@
+/*
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under the BSD-style license found in the
+ *  LICENSE file in the root directory of this source tree. An additional grant
+ *  of patent rights can be found in the PATENTS file in the same directory.
+ *
+ */
+
+#include <string>
+
+#include <osquery/core.h>
+#include <osquery/tables.h>
+
+#include "osquery/tables/system/windows/registry.h"
+
+namespace osquery {
+namespace tables {
+
+struct sdb {
+  std::string description;
+  unsigned long long installTimestamp;
+  std::string path;
+  std::string type;
+};
+
+QueryData genShims(QueryContext& context) {
+  QueryData results;
+  QueryData sdbResults;
+  QueryData shimResults;
+  std::map<std::string, sdb> sdbs;
+
+  queryKey("HKEY_LOCAL_MACHINE",
+           "SOFTWARE\\Microsoft\\Windows "
+           "NT\\CurrentVersion\\AppCompatFlags\\InstalledSDB",
+           sdbResults);
+  for (const auto& rKey : sdbResults) {
+    QueryData appResults;
+    sdb sdb;
+    std::string subkey = rKey.at("subkey");
+    std::string sdbId = subkey.substr(subkey.find("{"), subkey.length());
```

Comment:
If `subkey.find("{")` does not find a match the result is `std::string::npos` which is a very large number. It will cause bad things to happen.

Try 
```cpp
auto start = subkey.find("{");
if (start == std::string::npos) {
  continue;
}
```